docs/internals/signatures.rst

```diff
@@ -0,0 +1,153 @@
+
+Examples
+--------
+
+Because examples speak clearer than a thousand words.
+
+A function with no arguments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here is a sample of a function always returning the same value. Since it has
+no arguments, there is no need to supply any of the named
+
+.. code-block:: python
+
+    def constant_4():
+        """Return 4, always and ever."""
+        return 4
+
+.. code-block:: java
+
+    @org.python.Method(
+        __doc__ = "Return 4, always and ever."
+    )
+    public function org.python.Object constant_4() {
+        return org.python.types.Int(4);
+    }
+
+
+A function with two arguments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Another simple function is that of adding two given numbers.
+
+.. code-block:: python
+
+    def add(num1, num2):
+        """Add two numbers."""
+        return num1 + num2
+
+.. code-block:: java
+
+    @org.python.Method(
+        __doc__ = "Add two numbers.",
+        args = {"num1", "num2"}
+    )
+    public function org.python.Object add(org.python.Object num1, org.python.Object num2) {
+        // Left as exercise for the reader.
+    }
+
+
+A function with a default argument
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Similarly, we might want to make the second argument optional, allowing you
+to either add ``1`` to the number, or the supplied argument.
+
+.. code-block:: python
+
+    def inc(num, delta=1):
+        """Increment a number."""
+        return num + delta
+
+.. code-block:: java
+
+    @org.python.Method(
+        __doc__ = "Add two numbers.",
+        args = {"num", "delta"},
+        default_args = {"delta"}
+    )
+    public function org.python.Object inc(org.python.Object num, org.python.Object delta) {
+        if (delta == null) {
+            delta = new org.python.types.Int(1);
+        }
+        // Left as exercise for the reader.
+    }
+
+
+A function with variable arguments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Of course, sometimes you don't want to specify a specific number of arguments,
+but accept as many as you can get. For instance, the ``min`` function.
+
+.. code-block:: python
+
+    def min(first, *others):
+        """Get the minimum of the supplied arguments."""
+        val = first
+        for other in others:
+            if other < val:
+                val = other
+        return val
+
+.. code-block:: java
+
+    @org.python.Method(
+        __doc__ = "Get the minimum of the suppliend arguments.""",
+        args = {"first", "others"},
+        varargs = "others"
+    )
+    public function org.python.Object min(org.python.Object first, org.python.Object [] others) {
+        org.python.Object val = first;
+        for (other: others) {
+            if (other.__lt__(val)) {
+                val = other;
+            }
+        }
+        return val;
+    }
+
+A function accepting keyword arguments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Of course, sometimes you want to use keyword arguments. Unfortunately, that is
```

Comment:
The `kwargs` is not really used in the codebase, so I am a bit edgy before documenting how it should be used when there is no example to look at.